### ISSUE TITLE
[Feature request] New format string in translation file

### ISSUE THREAD

**Issue by FurryR** [DarwinBaker/AATool#79]
# Feature

TLDR: Supports `index-arg` and weak typing in translation file just like [`std::format`](https://en.cppreference.com/w/cpp/utility/format/format).

# Reason

I started to make Simplified Chinese translation for AATool, but I encountered some issues.
For example, `en-us` version:
```json
{
  "narrator.position.screen": "Screen element %s{1} out of %s{2}"
}
```
Can be translated into:
```json
{
  "narrator.position.screen": "第 %s{1} 个屏幕元素，共 %s{2} 个"
}
```
But this one is more natural:
```json
{
  "narrator.position.screen": "第 %s{2} 个中的 %s{1} 个屏幕元素"
}
```
(*`{order}` does not exist in current translation file, just for example*)
So I change the order of the arguments, but it is impossible to do that in current version.
I have also noticed that:
```json
{
  "narrator.position.object_list": "Selected row element %s out of %s",
  "narration.suggestion.tooltip": "Selected suggestion %d out of %d: %s (%s)",
}
```
Yes, `%s` and `%d` are confusing.
What do I want is:
```json5
{
  "narrator.position.screen": "Screen element {0} out of {1}" // no more strong typing, yes!
  // "narrator.position.screen": "第 {1} 个中的 {0} 个屏幕元素" // this looks great
}
```
So that I can make the translation easier to read.

**Comment by FurryR** [comment]
I will make a pull request in a few days.
If I do not, notify me.
